tinygrad/codegen/linearizer.py

```diff
@@ -102,12 +102,16 @@ def __init__(self, ast:LazyOp, output_buffer:LazyBuffer):
     self.ast = ast.src[0] if ast.op == MovementOps.RESHAPE else ast
 
     # get the output buffers
+    # after linearize, this is will be deduped by key=buf.realized
     self.bufs = [output_buffer] + dedup(ast.buffers)
 
+    # calculate mapping from to unique RawBuffer index
+    self.bufmap = [dedup([x.realized for x in self.bufs]).index(buf.realized) for buf in self.bufs]
```

Comment:
Use the exact same logic for this as ASTRunner.exec. RawConsts shouldn't be in the bufmap

Reply:
RawConsts are loaded in cstyle.py using the LOAD uop, so the bufmap needs to have RawConsts.

There are three classes of usage:

linearizer: LazyBuffers (RawBuffer and RawConst), LocalBuffers
cstyle/assembly: deduped LazyBuffers, LocalBuffers
input: deduped LazyBuffers without RawConsts.

The bufmap is for the cstyle/assembly to know which sources to read from.

Maybe I am missing something.